#!/usr/bin/env python
PACKAGE = "stereo_depth_correction"

from math import pi

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("trans_x_", double_t, 0, "x coord of translation", -0.0850857, -1.0, 1.0)
gen.add("trans_y_", double_t, 0, "y coord of translation", -0.06, -1.0, 1.0)
gen.add("trans_z_", double_t, 0, "z coord of translation", 0.03, -1.0, 1.0)
gen.add("roll_", double_t, 0, "roll angle", 0, -pi/2, pi/2)
gen.add("pitch_", double_t, 0, "pitch angle", -0.05, -pi/2, pi/2)
gen.add("yaw_", double_t, 0, "yaw angle", 0, -pi/2, pi/2)
gen.add("mr_BF", int_t, 0, "size of bilateral filter", 11, 1, 99)
gen.add("grid_S2D", int_t, 0, "grid size of SparseToDense", 4, 1, 98)

exit(gen.generate(PACKAGE, "stereo_depth_correction", "calibration"))
